pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -758,11 +791,14 @@ func (m *kubeGenericRuntimeManager) doPodResizeAction(pod *v1.Pod, podStatus *ku
 	}
 }
 
-func (m *kubeGenericRuntimeManager) updatePodContainerResources(pod *v1.Pod, resourceName v1.ResourceName, containersToUpdate []containerToUpdateInfo) error {
+func (m *kubeGenericRuntimeManager) updatePodContainerResources(pod *v1.Pod, resourceName v1.ResourceName, containersToUpdate []containerToUpdateInfo, isRestartableInitContainer bool) error {
 	klog.V(5).InfoS("Updating container resources", "pod", klog.KObj(pod))
 
 	for _, cInfo := range containersToUpdate {
 		container := pod.Spec.Containers[cInfo.apiContainerIdx].DeepCopy()
```

Comment:
Don't do this `DeepCopy()` if we're not using it.

Reply:
I guess we are using `container` in switch case below? https://github.com/kubernetes/kubernetes/pull/128367/files/76c5034c84071f7e4dafb85df4f309784cc08705#diff-0e1a8d22b0f1dc42d412749c6a916b4c76394ead739d4bfbf26564faedd74a2fR804-R823